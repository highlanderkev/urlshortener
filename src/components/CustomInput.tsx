import React from "react";

type InputType = "text" | "url";

type CustomInputProps = {
  type: InputType;
  name: string;
  placeholder: string;
  input: string;
  onInputChange: Function;
};

export default class CustomInput extends React.Component<CustomInputProps> {
  handleChange = (event: Event) => {
    this.props.onInputChange(event?.target?.value);
  };

  render() {
    return (
      <div className="form-floating">
        <input
          type={this.props.type}
          name={this.props.name}
          value={this.props.input}
          onChange={this.handleChange}
          placeholder={this.props.placeholder}
          className="form-control"
        />
        <label htmlFor={this.props.name}>{this.props.name}</label>
        <span>Value: {this.props.input}</span>
      </div>
    );
  }
}
