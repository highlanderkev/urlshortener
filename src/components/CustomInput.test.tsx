import React from 'react';
import CustomInput from "./CustomInput";
import renderer from "react-test-renderer";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()})
describe('Custom Input Component', () => {
  it("matches snapshot", () => {
    const customInput = renderer.create(<CustomInput type={"text"} name={"test"} placeholder={"test"} input={""} onInputChange={() => {}}/>);
    let tree = customInput.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays prop name by default", () => {
    const input = shallow(<CustomInput
      type={"url"}
      name={"url"}
      placeholder={"https://google.com"}
      input={""}
      onInputChange={() => {}}
    />)
    expect(input.text()).toEqual("url");
  });

  it("calls prop function with value on input change", () => {
    const mockFn = jest.fn();
    const input = shallow(<CustomInput
      type={"url"}
      name={"url"}
      placeholder={"https://google.com"}
      input={"test"}
      onInputChange={mockFn}
    />)
    input.find('input').simulate('change', { target: { value: 'test' } })
    expect(mockFn.mock.calls[0][0]).toBe('test');
  });
})

