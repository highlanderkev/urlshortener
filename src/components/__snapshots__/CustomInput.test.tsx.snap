// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input Component matches snapshot 1`] = `
<div
  className="form-floating"
>
  <input
    className="form-control"
    name="test"
    onChange={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
  <label
    htmlFor="test"
  >
    test
  </label>
</div>
`;
